src/LinkDotNet.NCronJob/RetryPolicies/PolicyCreatorFactory.cs

```diff
@@ -0,0 +1,15 @@
+namespace LinkDotNet.NCronJob;
+
+internal class PolicyCreatorFactory
+{
+    private readonly IServiceProvider serviceProvider;
+
+    public PolicyCreatorFactory(IServiceProvider serviceProvider) => this.serviceProvider = serviceProvider;
+
+    public TPolicyCreator Create<TPolicyCreator>() where TPolicyCreator : IPolicyCreator, new()
+    {
+        var creator = new TPolicyCreator();
+        (creator as IInitializablePolicyCreator)?.Initialize(serviceProvider);
```

Comment:
That would also reduce code here

Reply:
As above - no

Reply:
Same as above, can't add ctor to types used in generic parameters of attributes.This means that when a generic attribute is used, its construction needs to be fully "closed". https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/generic-attributes

Reply:
I can simplify the code and avoid the generic attributes, but then it won't be easy to extend it and allow users to make their own retry policies if they so choose. It comes down to flexibility on the user side vs simplicity on the library side.

Reply:
Nono - I am 100% onboard with your changes!